modules/microshift-oc-mirror-transform-imageset-to-crio.adoc

```diff
@@ -0,0 +1,87 @@
+//Module included in the following assemblies:
+//
+//* microshift_running_apps/microshift-operators-olm.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-oc-mirror-transform-imageset-to-crio_{context}"]
+= Configuring CRI-O for using registry mirror for Operators
+
+Use the following example procedure to transform the `imageContentSourcePolicy.yaml` created with the oc mirror tool into a setting you can add to the CRI-O container runtime configuration.
+
+.Prerequisites
+* The {oc-first} is installed.
+* Operator Lifecycle Manager (OLM) is installed.
+* The oc-mirror OpenShift CLI (oc) plugin is installed.
+* The yq binary is installed.
+* `ImageContentSourcePolicy` and `CatalogSource` YAML files are available in the `oc-mirror-workspace/results-*` directory.
+
+.Procedure
+
+. Confirm the contents of the `ImageContentSourcePolicy` YAML by running the following command:
++
+[source,terminal]
+----
+$ cat oc-mirror-workspace/<results-1707148826>/imageContentSourcePolicy.yaml <1>
+----
+<1> Replace _<results-1707148826>_ with the results directory name.
++
+.Example output
+[source,terminal]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  labels:
+    operators.openshift.org/catalog: "true"
+  name: operator-0
+spec:
+  repositoryDigestMirrors:
+  - mirrors:
+    - registry.<example.com>/amq7 <1>
+    source: registry.redhat.io/amq7
+----
+<1> Example name. Your registry name is listed here.
+
+. Transform the `imageContentSourcePolicy.yaml` into a format ready for CRI-O configuration by running the following command:
++
+[source,terminal]
+----
+yq '.spec.repositoryDigestMirrors[] as $item ireduce([]; . + [{"mirror": $item.mirrors[], "source": ($item | .source)}]) | .[] |
+  "[[registry]]
+      prefix = \"" + .source + "\"
+      location = \"" + .mirror + "\"
+      mirror-by-digest-only = true
+      insecure = true
+      "' ./icsp.yaml
+----
++
+.Example output
+[source,terminal]
+----
+[[registry]]
+      prefix = "registry.redhat.io/amq7"
+      location = "registry.example.com/amq7"
+      mirror-by-digest-only = true
+      insecure = true
+----
+
+. Add the output to the CRI-O configuration file in the `/etc/containers/registries.conf.d/` directory:
```

Comment:
I checked this and CRI-O doesn't watch the file and hot-reload it in case of change.
But it doesn't need to necessarily restarted, SIGHUP can be sent to the process instead of restarting:
```
$ sudo crictl pull something.that.doesnt.exist.com/microshift/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16
E0325 10:00:10.031441 2999538 remote_image.go:167] "PullImage from image service failed" err="rpc error: code = Unknown desc = pinging container registry something.that.doesnt.exist.com: Get \"https://something.that.doesnt.exist.com/v2/\": dial tcp: lookup something.that.doesnt.exist.com on 192.168.124.1:53: no such host" image="something.that.doesnt.exist.com/microshift/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16"
FATA[0000] pulling image: rpc error: code = Unknown desc = pinging container registry something.that.doesnt.exist.com: Get "https://something.that.doesnt.exist.com/v2/": dial tcp: lookup something.that.doesnt.exist.com on 192.168.124.1:53: no such host 

$ sudo vi /etc/containers/registries.conf
# Added following:
[[registry]]
      prefix = "something.that.doesnt.exist.com/microshift/busybox"
      location = "quay.io/microshift/busybox"
      mirror-by-digest-only = false

$ sudo systemctl status crio | grep PID
   Main PID: 2967754 (crio)

$ sudo kill -SIGHUP 2967754

$ sudo crictl pull something.that.doesnt.exist.com/microshift/busybox@sha256:b5d6fe0712636ceb7430189de28819e195e8966372edfc2d9409d79402a0dc16
Image is up to date for quay.io/microshift/busybox@sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5

```

Reply:
@pmtk Do we want users to execute the commands beginning with `sudo systemctl status crio | grep PID`, or restart crio?

Reply:
Maybe restart crio for simplicity? I don't think it would kill the containers, just restart the daemon so should be good enough